src/WebJobs.Script.WebHost/IFileMonitoringService.cs

```diff
@@ -13,6 +13,6 @@ public interface IFileMonitoringService : IHostedService
 
         void ResumeRestart();
 
-        Task ScheduleRestartAsync(bool shutdown);
+        Task RestartIfScheduledAsync();
```

Comment:
First, it looks like the ResumeRestart method above is never called anywhere. Can't it be removed? Next, rather than adding this new method, how about just having SuspendRestart take a bool autoResume parameter that when true, will do this "restart if scheduled" logic internally? Likely it'll be the case that anytime you suspend, you want to "restart if scheduled" after, so why not bake it into the disposal pattern to prevent errors and simplify things?

Reply:
Regarding ResumeRestart there was the discussion about the method:
https://github.com/Azure/azure-functions-host/pull/6799#discussion_r509661565

Regarding autoResume parameter
it will require to call async from sync Dispose() what it's not good.
There is DisposeAsync() in C# 8.0 but we still use C# 7.2.